### ISSUE TITLE
ns3/mmwave-error-model.h missing file

### ISSUE THREAD

**Issue by BrenoMelo** [signetlabdei/millicar#13]
Hello,

I am trying to use millicar module in NS-3, but I got this issue:

`[1882/2578] Compiling src/millicar/model/mmwave-sidelink-phy.cc
[1883/2578] Compiling src/millicar/model/mmwave-sidelink-mac.cc`

`In file included from ../src/millicar/model/mmwave-sidelink-spectrum-phy.cc:30:
../src/millicar/model/mmwave-sidelink-spectrum-phy.h:43:10: fatal error: ns3/mmwave-error-model.h: No such file or directory
   43 | #include <ns3/mmwave-error-model.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~`
`compilation terminated.`

`In file included from ../src/millicar/model/mmwave-sidelink-phy.h:24,
                 from ../src/millicar/model/mmwave-sidelink-phy.cc:21:
../src/millicar/model/mmwave-sidelink-spectrum-phy.h:43:10: fatal error: ns3/mmwave-error-model.h: No such file or directory
   43 | #include <ns3/mmwave-error-model.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.`
`./src/millicar/model/mmwave-sidelink-mac.cc: In member function ‘ns3::mmwave::SlotAllocInfo ns3::millicar::MmWaveSidelinkMac::ScheduleResources(ns3::mmwave::SfnSf)’:
../src/millicar/model/mmwave-sidelink-mac.cc:224:43: error: ‘class ns3::mmwave::MmWaveAmc’ has no member named ‘CalculateTbSize’
  224 |     uint32_t availableBytesPerLc = m_amc->CalculateTbSize(mcs, availableSymbolsPerLc);
      |                                           ^~~~~~~~~~~~~~~`
`./src/millicar/model/mmwave-sidelink-mac.cc:241:39: error: ‘class ns3::mmwave::MmWaveAmc’ has no member named ‘GetMinNumSymForTbSize’
  241 |     uint32_t assignedSymbols = m_amc->GetMinNumSymForTbSize (assignedBytes, mcs);
      |                                       ^~~~~~~~~~~~~~~~~~~~~
`
`../src/millicar/model/mmwave-sidelink-mac.cc: In member function ‘void ns3::millicar::MmWaveSidelinkMac::DoSlSinrReport(const ns3::SpectrumValue&, uint16_t, uint8_t, uint32_t)’:
../src/millicar/model/mmwave-sidelink-mac.cc:488:83: error: no matching function for call to ‘ns3::mmwave::MmWaveAmc::CreateCqiFeedbackWbTdma(const ns3::SpectrumValue&, uint8_t&)’
  488 |     m_slCqiReported.at (rnti).push_back (m_amc->CreateCqiFeedbackWbTdma (sinr, mcs));
      |                                                                                   ^`
`In file included from ../src/millicar/model/mmwave-sidelink-mac.cc:21:
./ns3/mmwave-amc.h:59:7: note: candidate: ‘int ns3::mmwave::MmWaveAmc::CreateCqiFeedbackWbTdma(const ns3::SpectrumValue&, uint8_t, uint32_t, int&)’
   59 |   int CreateCqiFeedbackWbTdma (const SpectrumValue& sinr, uint8_t numSym, uint32_t tbs, int &mcsWb);
      |       ^~~~~~~~~~~~~~~~~~~~~~~`
`./ns3/mmwave-amc.h:59:7: note:   candidate expects 4 arguments, 2 provided
../src/millicar/model/mmwave-sidelink-mac.cc:493:65: error: no matching function for call to ‘ns3::mmwave::MmWaveAmc::CreateCqiFeedbackWbTdma(const ns3::SpectrumValue&, uint8_t&)’
  493 |     cqiTemp.push_back (m_amc->CreateCqiFeedbackWbTdma (sinr, mcs));
      |                                                                 ^`
`In file included from ../src/millicar/model/mmwave-sidelink-mac.cc:21:
./ns3/mmwave-amc.h:59:7: note: candidate: ‘int ns3::mmwave::MmWaveAmc::CreateCqiFeedbackWbTdma(const ns3::SpectrumValue&, uint8_t, uint32_t, int&)’
   59 |   int CreateCqiFeedbackWbTdma (const SpectrumValue& sinr, uint8_t numSym, uint32_t tbs, int &mcsWb);
      |       ^~~~~~~~~~~~~~~~~~~~~~~
./ns3/mmwave-amc.h:59:7: note:   candidate expects 4 arguments, 2 provided`

**Comment by MatteoDrago** [comment]
Hi @BrenoMelo, I just tried to install it from scratch and it compiled correctly. 
I suggest you to update the ns3-mmwave version that you are using. Let me know if you have any further problems!

**Comment by BrenoMelo** [comment]
Hi @MatteoDrago, thanks for your fast reply!

I am using this ns3-mmwave version: https://github.com/nyuwireless-unipd/ns3-mmwave

What I did:

1. Downloaded `ns3-mmwave`
2. I ran this command: `./waf configure --disable-python --enable-examples && ./waf build`
3. Downloaded and extracted `millicar` to `ns3-mmwave/src` folder
4. I ran this command again: `./waf configure --disable-python --enable-examples` (executed without any error)
5. `./waf build` (I got the reported error)
